src/Identifier/LdapIdentifier.php

```diff
@@ -0,0 +1,243 @@
+use Cake\ORM\Entity;
+use ErrorException;
+use InvalidArgumentException;
+use RuntimeException;
+
+/**
+ * LDAP Identifier
+ *
+ * Identifies authentication credentials using LDAP.
+ *
+ * ```
+ *  new LdapIdentifier([
+ *       'host' => 'ldap.example.com',
+ *       'bindDN' => function($username) {
+ *           return $username; //transform into a rdn or dn
+ *       },
+ *       'options' => [
+ *           LDAP_OPT_PROTOCOL_VERSION => 3
+ *       ]
+ *  ]);
+ * ```
+ *
+ * @link https://github.com/QueenCityCodeFactory/LDAP
+ */
+class LdapIdentifier extends AbstractIdentifier
+{
+
+    /**
+     * Default configuration
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'ldapClass' => null,
+        'fields' => [
+            'username' => 'username',
+            'password' => 'password'
+        ],
+        'port' => 389
+    ];
+
+    /**
+     * List of errors
+     *
+     * @var array
+     */
+    protected $_errors = [];
+
+    /**
+     * LDAP connection object
+     *
+     * @var \Authentication\Identifier\LdapInterface
+     */
+    protected $_ldap = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        $this->_checkLdapExtension();
+        $this->_checkLdapConfig();
+        $this->_buildLdapObject();
+    }
+
+    /**
+     * Checks if the php LDAP extension is loaded
+     *
+     * @throws \RuntimeException
+     * @return void
+     */
+    protected function _checkLdapExtension()
+    {
+        if (!extension_loaded('ldap')) {
+            throw new RuntimeException('You must enable the ldap extension to use the LDAP identifier.');
+        }
+    }
+
+    /**
+     * Checks the LDAP config
+     *
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
+     * @return void
+     */
+    protected function _checkLdapConfig()
+    {
+        if (!defined('LDAP_OPT_DIAGNOSTIC_MESSAGE')) {
+            define('LDAP_OPT_DIAGNOSTIC_MESSAGE', 0x0032);
+        }
+        if (!isset($this->_config['bindDN'])) {
+            throw new RuntimeException('Config `bindDN` is not set.');
+        }
+        if (!is_callable($this->_config['bindDN'])) {
+            throw new InvalidArgumentException(sprintf(
+                'The `bindDN` config is not a callable. Got `%s` instead.',
+                gettype($this->_config['bindDN'])
+            ));
+        }
+        if (!isset($this->_config['host'])) {
+            throw new RuntimeException('Config `host` is not set.');
+        }
+    }
+
+    /**
+     * Constructs the LDAP object and sets it to the property
+     *
+     * @throws \RuntimeException
+     * @return void
+     */
+    protected function _buildLdapObject()
+    {
+        if (empty($this->_config['ldapClass'])) {
+            $this->setConfig('ldapClass', Ldap::class);
+        }
+
+        if (is_string($this->_config['ldapClass'])) {
```

Comment:
If you're trying to allow dependency injection for testing, why not pass the object|null instead of a class name? Right now if a different implementation can't have any constructor arguments which can be a bit limiting and doesn't play nice with PHPUnit mocks.